### PR TITLE
Misc PDA code improvements

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

* Removes unnecessary support for the now-removed job disks from Tablet's TGUI menu, and tablet's ui_act.
* Adds autodoc comments to computer files
* Removes the unused 'unsendable' var on computer files
* Generally improves code on tablets, now process isn't looping through every idle thread twice!
* Moves the check for program in idle_threads above checking if supported by hardware, because it's already running, so there's no need to check.

## Why It's Good For The Game

Every time I look at PDA's code, I am disappointed. I started the hardware removal PRs to make it a little less depressing to look at, but this is just a more general cleanup with no specific target.

## Changelog

Not necessary

### CODE REVIEW

File: code/modules/modular_computers/computers/item/computer.dm [hunk 1]
```diff
@@ -509,37 +507,31 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
...
 	else
 		data["PC_batteryicon"] = "batt_5.gif"
 		data["PC_batterypercent"] = "N/C"
-		data["PC_showbatteryicon"] = battery_module ? 1 : 0
 
 	switch(get_ntnet_status())
 		if(NTNET_NO_SIGNAL)
 			data["PC_ntneticon"] = "sig_none.gif"
-		if(NTNET_LOW_SIGNAL)
+		else if(NTNET_LOW_SIGNAL)
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/modules/modular_computers/computers/item/computer.dm:517]: right i knew that yeah of course

File: code/modules/research/ordnance/_scipaper.dm [hunk 2]
```diff
@@ -113,9 +113,9 @@
  * Things to check: tier, gain, and partner here. ex_path and record datums in subtypes.
  */
 /datum/scientific_paper/proc/allowed_to_publish(datum/techweb/techweb_to_check)
-	if(!tier || !gains || !partner_path || (0 in gains))
+	if(!tier || !gains || !partner_path || !gains.len)
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/modules/research/ordnance/_scipaper.dm:116]: man i dont know but it's a list

[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/modules/research/ordnance/_scipaper.dm:116]: actually probably

File: code/modules/research/ordnance/_scipaper.dm [hunk 3]
```diff
@@ -206,16 +206,10 @@
...
-	. = ..()
+	return ..()
 
 /datum/scientific_paper/explosive/set_experiment(ex_path = null, variable = null, data = null)
-	var/invalid = FALSE
-
-	invalid = invalid || !ispath(ex_path, /datum/experiment/ordnance/explosive)
-	invalid = invalid || !variable
-	invalid = invalid || !istype(data, /datum/data/tachyon_record)
-
```
Comment (ShizCalev) [inline code/modules/research/ordnance/_scipaper.dm:217]: what the fuck

**Pull request merged**